docs/troubleshooting.md

```diff
@@ -9,19 +9,56 @@ menuWeight: 125
 
 # Dispatcher
 
-The Mesos cluster dispatcher is responsible for queuing, tracking, and supervising drivers. Potential problems may arise if the dispatcher does not receive the resources offers you expect from Mesos, or if driver submission is failing. To debug this class of issue, visit the Mesos UI at `http://<dcos-url>/mesos/` and navigate to the sandbox for the dispatcher.
+*   The Mesos cluster dispatcher is responsible for queuing, tracking, and supervising drivers. Potential problems may
+    arise if the dispatcher does not receive the resources offers you expect from Mesos, or if driver submission is
+    failing. To debug this class of issue, visit the Mesos UI at `http://<dcos-url>/mesos/` and navigate to the sandbox
+    for the dispatcher.
+
+*   Spark has an internal mechanism for detecting the IP of the host. We use this method by default, but sometimes it
+    fails, returning errors like these:
+
+    ```
+    ERROR SparkUncaughtExceptionHandler: Uncaught exception in thread Thread[main,5,main]
+        java.net.UnknownHostException: ip-172-31-4-148: ip-172-31-4-148: Name or service not known
+            at java.net.InetAddress.getLocalHost(InetAddress.java:1505)
+            at org.apache.spark.util.Utils$.findLocalInetAddress(Utils.scala:891)
+            at org.apache.spark.util.Utils$.org$apache$spark$util$Utils$$localIpAddress$lzycompute(Utils.scala:884)
+            at org.apache.spark.util.Utils$.org$apache$spark$util$Utils$$localIpAddress(Utils.scala:884)
+            at org.apache.spark.util.Utils$$anonfun$localHostName$1.apply(Utils.scala:941)
+            at org.apache.spark.util.Utils$$anonfun$localHostName$1.apply(Utils.scala:941)
+            at scala.Option.getOrElse(Option.scala:121)
+            at org.apache.spark.util.Utils$.localHostName(Utils.scala:941)
+            at org.apache.spark.deploy.mesos.MesosClusterDispatcherArguments.<init>(MesosClusterDispatcherArguments.scala:27)
+            at org.apache.spark.deploy.mesos.MesosClusterDispatcher$.main(MesosClusterDispatcher.scala:103)
+            at org.apache.spark.deploy.mesos.MesosClusterDispatcher.main(MesosClusterDispatcher.scala)
+        Caused by: java.net.UnknownHostException: ip-172-31-4-148: Name or service not known
+            at java.net.Inet6AddressImpl.lookupAllHostAddr(Native Method)
+            at java.net.InetAddress$2.lookupAllHostAddr(InetAddress.java:928)
+            at java.net.InetAddress.getAddressesFromNameService(InetAddress.java:1323)
+            at java.net.InetAddress.getLocalHost(InetAddress.java:1500)
+            ... 10 more
+    18/01/25 17:42:57 INFO ShutdownHookManager: Shutdown hook called
+    ```
+
+    In this case, remove the environment variable called `NO_BOOTSTRAP` from the Dispatcher config, either via the UI, editing the task or set to false in the options.json, and restart the
```

Comment:
I don't think you can set this in the options.json, at least when installing. Would it be through an "update" command?